diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LongNumericalValuesSeparatedRule.kt

```diff
@@ -0,0 +1,139 @@
+                    fixConstant(node, configuration)
+                }
+            }
+        }
+
+    }
+
+    private fun fixConstant(node: ASTNode, configuration: LongNumericalValuesConfiguration) {
+        val parts = node.text.split(".")
+        val realPart = removePrefixSuffix(parts[0])
+
+        val resultRealPart = StringBuilder(nodePrefix(node.text))
+
+        // We can get here in 2 cases:
+        // 1. When integer number is > maxLength
+        // 2. When float number realPart > maxLength
+        if (realPart.length > configuration.maxLength) {
+            val chunks = realPart.reversed().chunked(configuration.maxBlockLength).reversed()
+            resultRealPart.append(chunks.joinToString(separator = "_") { it.reversed() })
+        } else {
+            // Here we get in 1 case: when realPart of float number is < maxLength
+            resultRealPart.append(realPart).append(".")
+        }
+
+        if (parts.size > 1 && parts[1].length > configuration.maxLength) {
+            val resultFractionalPart = StringBuilder()
+            val chunks = parts[1].reversed().chunked(configuration.maxBlockLength).reversed()
+            resultFractionalPart.append(chunks.joinToString(separator = "_") { it.reversed() })
+            resultFractionalPart.append(nodeSuffix(node.text))
+
+            (node as LeafPsiElement).replaceWithText(resultRealPart.append(resultFractionalPart).toString())
+            return
+        }
+
+        resultRealPart.append(nodeSuffix(node.text))
+        (node as LeafPsiElement).replaceWithText(resultRealPart.toString())
+    }
+
+    private fun nodePrefix(nodeText: String) : String {
+        if (nodeText.startsWith("0b"))
+            return "0b"
+
+        if(nodeText.startsWith("0x"))
+            return "0x"
+
+        return ""
+    }
+
+    private fun nodeSuffix(nodeText: String) : String {
+        if (nodeText.endsWith("L"))
+            return "L"
+
+        if (nodeText.endsWith("f", true))
+            return "f"
+
+        return ""
+    }
+
+
+    private fun isValidConstant(text: String, configuration: LongNumericalValuesConfiguration, node: ASTNode) : Boolean {
+        if (text.contains("_")) {
+            val number = removePrefixSuffix(text)
+            checkBlocks(number, configuration, node)
+            return true
+        }
+
+        val parts = text.split(".")
+
+        val realPart = removePrefixSuffix(parts[0])
+
+
+        if (parts.size > 1) {
+            return parts[1].length < configuration.maxLength
+        }
+
+        return realPart.length < configuration.maxLength
+    }
+
+    private fun checkBlocks(text: String, configuration: LongNumericalValuesConfiguration, node: ASTNode) {
+        val blocks = text.split("_", ".")
+
+        blocks.forEach {
+            if (it.length > configuration.maxBlockLength) {
+                Warnings.LONG_NUMERICAL_VALUES_SEPARATED.warn(configRules, emitWarn, isFixMode, "this block is too long $it", node.startOffset)
```

Comment:
```suggestion
                Warnings.LONG_NUMERICAL_VALUES_SEPARATED.warn(configRules, emitWarn, false, "this block is too long $it", node.startOffset)
```
I  believe with this argument `LintError` will be marked as 'cannot be auto-corrected'.